### ISSUE TITLE
Shap loss for regression problems

### ISSUE THREAD

**Issue by mschiwek** [shap/shap#1784]
Hi,

in the paper "Explainable AI for Trees: From Local Explanations to Global Understanding" are monitoring plots, which explain the shap loss values for squared errors. If I understand it correctly the hospital duration model solves a regression problem. As a result, I couldn't find out how these shap loss values are calculated. If I'm not mistaken, currently you can only calculate these values for classification problems using the shap library.

Have I misunderstood something or is there an easy way to calculate these values for regression problems?

**Comment by mschiwek** [comment]
I did some research in the code. It seems like we are able to explain squared_loss. What confuses me is the parameter model_output, which has to be set to log_loss.

explainer = TreeExplainer(model, x, feature_dependence="independent", model_output="logloss")

By setting it to logloss I was able to explain my squared_error. What I do not undestand is why the model_output is set to logloss. Isn't logloss usually used for classification problems?

**Comment by alberto-bracci** [comment]
hi, I am having a similar problem. I am using a xgboost regressor with reg:logistic objective. I am using the same options as you, but shap values + expected values do not sum up to the squared error (by far) nor by the log loss computed as if it was a classification problem (by little). 

What do you exactly mean by "explain my squared error" ?

**Comment by sebastianptz** [comment]
Hi, I am running into the same problem using the lightgbm regressor. Did you figure this out by now?

**Comment by CoteDave** [comment]
Hi, same here, is it possible to calculate the shap loss for regession problem ?
